substrate/primitives/runtime/src/generic/unchecked_extrinsic.rs

```diff
@@ -545,13 +628,41 @@ impl<'a, Address: Decode, Signature: Decode, Call: DecodeWithMemTracking, Extens
 	}
 }
 
+/// Something which holds the actual call and maybe its encoded form.
+pub struct CallAndMaybeEncoded<T> {
+	encoded: Option<Vec<u8>>,
+	call: T,
+}
+
+impl<T> CallAndMaybeEncoded<T> {
+	/// Converts `self` into the underlying call.
+	pub fn into_call(self) -> T {
+		self.call
+	}
+}
+
+impl<T> From<T> for CallAndMaybeEncoded<T> {
+	fn from(value: T) -> Self {
```

Comment:
This is to make it a bit backwards compatible since any `Call` type can be converted to a `CallAndMaybeEncoded`?